src/settings.cpp

```diff
@@ -1519,9 +1566,12 @@ static void SaveVersionInConfig(IniFile *ini)
 	char version[9];
 	seprintf(version, lastof(version), "%08X", _openttd_newgrf_version);
 
-	const char * const versions[][2] = {
+	std::string ini_version = std::to_string(INIFILE_VERSION);
```

Comment:
Not that it is your fault, but sometimes I think preventing "duplicate" code goes a bit too far. In this case someone decided to construct a table and then write a loop to copy the data from that table into the settings, just to prevent two (now three) similar looking lines.

What would the consensus be for something something like the following instead of the current ~15 lines?
```
IniGroup *group = ini->GetGroup("version");
group->GetItem("version_string", true)->SetValue(_openttd_revision);
group->GetItem("version_number", true)->SetValue(fmt::format("{:08X}", _openttd_newgrf_version));
group->GetItem("ini_version", true)->SetValue(std::to_string(INIFILE_VERSION));
```
Yes it is duplicating some code, but preventing a lot more boilerplate for setting up the table and looping over it.